### DISCUSSION TITLE
[Question] How to open and decode a string line by line when it was before encrypted

### DISCUSSION THREAD

**Discussion by Piioo** [rocketjob/symmetric-encryption#160] [General]
### Environment

Provide at least:
* ruby 2.7.6p219
* symmetric-encryption (4.5.0)
* Rails 6

### Expected Behavior

When I encrypt a string with `SymmetricEncryption.encrypt` then it should be decrypted by `SymmetricEncryption::Reader.open` line by line too. 
https://encryption.rocketjob.io/ -> Large File Encryption

### Actual Behavior
This code is not working, when I only pass the encrypted string into the `open` method
```ruby
foo = SymmetricEncryption.encrypt('foo', random_iv: false, compress: false, type: :string)
foo = StringIO.new(foo)
SymmetricEncryption::Reader.open(foo) do |file|
  file.each_line do |line|
     puts '###############'
     puts line
  end
end

# OpenSSL::Cipher::CipherError: bad decrypt
```

But it works when I `Base64.decode64` the string before. But does the line by line have any sense in large files, when we have first to decode64 it?
```ruby
foo = SymmetricEncryption.encrypt('foo', random_iv: false, compress: false, type: :string)
foo = StringIO.new(Base64.decode64(foo))
SymmetricEncryption::Reader.open(foo) do |file|
  file.each_line do |line|
     puts '###############'
     puts line
  end
end

# ###############
# foo
```

**Comment by reidmorrison** [comment]
Yes, in files we don't typically perform base 64 encoding, whereas storing encrypted data in the database we usually convert it to base 64 before saving. If you want to save the encrypted data in a binary column, you can set the encoding option to `:none`. Not recommended since we tried it in a proof of concept and ran into several challenges.
